enumerables.rb

```diff
@@ -0,0 +1,219 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    if is_a? Array
+      i = 0
+      while i < size
+        yield(self[i])
+        i += 1
+      end
+    elsif is_a? Hash
+      while i < size
+        yield(values[i], keys[i])
+        i += 1
+      end
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    arr_to_work = self
+    if arr_to_work.is_a? Array
+      i = 0
+      while i < arr_to_work.length
+        yield(arr_to_work[i], i)
+        i += 1
+      end
+    elsif arr_to_work.is_a? Hash
+      j = 0
+      while j < arr_to_work.length
+        yield([arr_to_work.keys[j], arr_to_work.values[j]], j)
+        j += 1
+      end
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    arr_to_work = self
+    if is_a? Array
+      myarr = []
+      arr_to_work.my_each do |item|
+        myarr.push(item) if yield(item)
+      end
+    elsif is_a? Hash
+      myarr = {}
+      arr_to_work.my_each do |k, v|
+        myarr[k] = v if yield(k, v)
+      end
+    end
+    myarr
+  end
+
+  def my_all?(pattern = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
```

Comment:
**[REQUIRED]**
* When no block or argument is given returns true when none of the collection members are false or nil.
![all](https://user-images.githubusercontent.com/44072711/68618912-5a9cff80-0498-11ea-928f-671cdb9f2ff6.png)